snapcraft/internal/errors.py

```diff
@@ -169,7 +171,7 @@ class tries to make them a bit more understandable.
                 error.validator_value['validation-failure'].format(error))
 
         if not messages:
-            messages.append(error.message)
+            messages.append(_patternProperties(error) or error.message)
 
```

Comment:
The [bug report](https://bugs.launchpad.net/snapcraft/+bug/1578024) proposed some much friendlier wording. How about a dictionary of messages based on the property? Perhaps using the generic one as a fallback.

> Issues while validating snapcraft.yaml: The 'parts' property does not match the required schema: Part names may only contain lowercase letters, numerals, and hyphens.